eno-ws/src/main/java/fr/insee/eno/ws/controller/GenerationController.java

```diff
@@ -0,0 +1,151 @@
+package fr.insee.eno.ws.controller;
+
+import fr.insee.eno.legacy.parameters.*;
+import fr.insee.eno.ws.PassePlat;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.MediaType;
+import org.springframework.http.codec.multipart.FilePart;
+import org.springframework.http.server.reactive.ServerHttpRequest;
+import org.springframework.http.server.reactive.ServerHttpResponse;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+import reactor.core.publisher.Mono;
+
+@Tag(name = "Generation of questionnaire")
+@Controller
+@RequestMapping("/questionnaire")
+@Slf4j
+@SuppressWarnings("unused")
+public class GenerationController {
+	
+	private final PassePlat passePlat;
+
+	public GenerationController(PassePlat passePlat) {
+		this.passePlat = passePlat;
+	}
+
+	@Operation(
+			summary = "Generation of questionnaire according to parameters.",
+			description = "Generate a questionnaire using the parameters file (required), metadata file (optional) " +
+					"and the specificTreatment file (optional). To use it, you have to upload all necessary files.")
+	@PostMapping(value = "in-2-out", 
+			produces = MediaType.APPLICATION_OCTET_STREAM_VALUE, consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
+	public Mono<Void> generate(
+			@RequestPart(value="in", required=true) Mono<FilePart> in,
```

Comment:
required=true by default

Reply:
indeed, I did some clean up for this afterwards